### ISSUE TITLE
ioc pkg: stripped down jail uses user set defaults

### ISSUE THREAD

**Issue by igalic** [iocage/libiocage#558]
`ioc pkg` runs a strip-down version of a jail

however, it appears that the ***set*** `defaults` for `exec_prestart` and `exec_start` are used.

We set some defaults which make this difficult:

```sh-session
root@container-host1 ~# ioc get all defaults | grep exec_
exec_stop:/bin/sh /etc/rc.shutdown
exec_timeout:600
exec_created:-
exec_jail_user:root
exec_prestart:set -eu ; mkdir -p "${IOCAGE_JAIL_PATH}/etc/facter/facts.d/" ; echo role=$IOCAGE_USER_FACTS_ROLE > "${IOCAGE_JAIL_PATH}/etc/facter/facts.d/role.txt"
exec_start:/bin/sh /etc/rc && /usr/local/bin/puppet apply --debug /usr/local/etc/puppet/environments/production/manifests/site.pp
exec_clean:yes
exec_poststop:-
exec_prestop:-
exec_fib:1
exec_poststart:-
root@container-host1 ~# 
```

ENV isn't set:

```
  [-] JailLaunch@webirc_pkg: FAILED [1.602s]
Executing (interactive): /usr/sbin/jail -c ip4.addr= ip4.saddrsel=1 ip4=new ip6.addr= ip6.saddrsel=1 ip6=new name=default-webirc_pkg host.hostname=webirc_pkg host.domainname=local path=/jails/jails/webirc/root securelevel=2 host.hostuuid=webirc_pkg devfs_ruleset=4 enforce_statfs=2 children.max=0 allow.set_hostname=1 allow.sysvipc=0 exec.prestart="/jails/jails/webirc/launch-scripts/prestart.sh" exec.prestop="/jails/jails/webirc/launch-scripts/prestop.sh" exec.poststop="/jails/jails/webirc/launch-scripts/poststop.sh" exec.jail_user=root sysvmsg=new sysvsem=new sysvshm=new allow.raw_sockets=0 allow.chflags=0 allow.mount=0 allow.mount.devfs=0 allow.mount.nullfs=0 allow.mount.procfs=0 allow.mount.fdescfs=0 allow.mount.zfs=0 allow.quotas=0 allow.socket_af=0 exec.timeout=600 stop.timeout=30 mount.fstab=/jails/jails/webirc/fstab mount.devfs=1 mount.fdescfs=0 allow.mount.tmpfs=0 allow.dying nopersist exec.poststart="/jails/jails/webirc/launch-scripts/host_command.sh" command=/usr/bin/true
  /jails/jails/webirc/launch-scripts/prestart.sh: IOCAGE_USER_FACTS_ROLE: parameter not set
  jail: /bin/sh -c "/jails/jails/webirc/launch-scripts/prestart.sh": failed
Jail webirc_pkg command failed.
Launching jail webirc_pkg failed
```

and of course, puppet doesn't exist, so that'll make rc startup difficult ;)

**Comment by igalic** [comment]
how did it take me so long to make this connection?

after all it was with @gronke's help at i fixed my `ioc pkg` failure before by setting `ioc set exec_timeout=600 defaults`

i'm not sure what the best way around this issue is, other than a separate, xor immutable `pkg_defaults`…

**Comment by igalic** [comment]
(so, i know how to fix this for myself, but i'd rather figure out how to fix it for future users who aren't me)

**Comment by igalic** [comment]
solved with #559
